### DISCUSSION TITLE
Problem with missing files

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#5007] [Q&A]
I have a channel (latest version of MC) where I'm reading a file from an sFTP server and writing it to a smb connected server. I'm using the attachment handler (set to Entire Message), Properties set to text/plain.  Both source and destination file type is set to binary and Set Data Types are all set to raw.  The files are PDFs.  It moves about 7500 files at one time during the day.

My problem is, when I check the smb folder, some files are not there even though in Mirth, looking at the message response, I see this:

SENT: File successfully written: smb://epic-nas.et1041.epichosted.com/dynamic/NavicureStatements/STMT_PRD_40052_IMPORT_20880702_4485956.pdf

Any suggestions as to why this is happening? I've verified that the process that uses these files does not run until well after the downloads are completed.  Kind of at a loss.

**Comment by pacmano1** [comment]
Did you solve this?   If not, did you try "Create Temp File" on the SMB writer?

**Comment by rutledj** [comment]
I am using the temp file option. Still seems to be an issue but I can't positively rule out some other process causing issues with the files. Still investigating.

**Reply by pacmano1** [reply]
You might try natively mounting the file system at the OS level and see if the same thing persists.   If yes, I would configure the file system to do audit logging.   Native mount or not, auditing should help you identify the problem.

**Reply by tonygermano** [reply]
I don't expect this to fix your issue, but the appropriate mime time is `application/pdf`, not `text/plain`. If nothing else, this should allow you to open the attachments in the mirth message browser with the integrated pdf viewer.
